### PR TITLE
Feature/swagger bugfixes

### PR DESCRIPTION
**Pull request by ben-vr**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "lint"
   ],
   "dependencies": {
+    "@types/node": "^10.12.18",
```
Comment (knor-el-snor) [inline package.json]: Shouldn't be needed I think

File: src/lib/swagger.ts [hunk 2]
```diff
@@ -2,23 +2,67 @@ import { Application } from 'express';
...
 
 /**
  * Serve swagger documentation
  */
 export function setSwagger(app: Application, route: string, filePath: string, options: SwaggerOptions = {}): void {
   try {
-    // Load yaml document
-    const swaggerDocument = yaml.safeLoad(fs.readFileSync(filePath, 'utf8'));
-    Object.assign(swaggerDocument, options);
+    const stats = fs.lstatSync(filePath);
```
Comment (knor-el-snor) [inline src/lib/swagger.ts:12]: Is it necessary to load stats when options.concatenate is not enabled? Shouldn't we move this into the if statement

File: tests/assets/almostValidDocsFolder/index.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+basePath: "/v2"
+tags:
+- name: "pet"
+  description: "Everything about your Pets"
+  externalDocs:
+    description: "Find out more"
+    url: "http://swagger.io"
+schemes:
+- "http"
+paths:
```
Comment (knor-el-snor) [inline tests/assets/almostValidDocsFolder/index.yml]: Why do you need to have `paths:` here? Can't you append this in code?

**Review by knor-el-snor — CHANGES_REQUESTED** [review]

File: src/lib/swagger.ts (hunk 2 above)
Comment (ben-vr) [reply src/lib/swagger.ts:12]: if (options.concatenate) {
      // Throw error if concatenate = true and filepath = file
      if (stats.isFile()) {
        throw new Error('Boolean concatenate cannot be true when you specify a file. When you want to concatenate, specify a folder');
      }
      if (stats.isDirectory()) {
        const swaggerContent = buildSwaggerDocumentFromFiles(filePath);
        swaggerDocument = yaml.safeLoad(swaggerContent);
      }
    } else {
      if (stats.isFile()) {
        // Load yaml document
        swaggerDocument = yaml.safeLoad(fs.readFileSync(filePath, 'utf8'));
        Object.assign(swaggerDocument, options);
      }
      if (stats.isDirectory()) {
        throw new Error('To concatenate a folder of swagger YMLS, you need to explicitly set the boolean concatenate on true for the swaggerOptions');
      }
    }

Yes because i throw an Error when the file is a directory and concatenate is not set

File: src/lib/express.ts [hunk 4]
```diff
@@ -63,5 +63,5 @@ export interface BodyParserOptions {
 }
 
 export interface RedisOptions extends ClientOpts {
-  client?: RedisClient;
+  client?: RedisClient | any;
```
Comment (knor-el-snor) [inline src/lib/express.ts]: Why is this needed?

**Review by knor-el-snor — APPROVED** [review]
